src/Translator.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace atk4\core;
+
+class Translator implements TranslatorInterface
+{
+    use ConfigTrait;
+
+    /**
+     * ISOCode of the main language.
+     *
+     * @var string
+     */
+    public $language;
+
+    /**
+     * ISOCode of the fallback language.
+     *
+     * @var string
+     */
+    public $fallback;
+
+    /**
+     * Array where Translation will be stored.
+     *
+     * @var array
+     */
+    protected $translations = [];
+
+    /**
+     * Raise exception if format of translation is not correct
+     *
+     * @var bool
+     */
+    public $raise_bad_format_exception = false;
+
+    /**
+     * Set primary language ISO Code
+     *
+     * @param string      $language
+     *
+     * @param string|null $fallback
+     *
+     * @return string
+     */
+    public function setLanguage(string $language,?string $fallback = null)
+    {
+        $this->language = $language;
+        $this->fallback = $fallback ?? false;
```

Comment:
Maybe change this to 
```php
$this->fallback = $fallback ?? $this->fallback ?? false;
```

Reply:
correct and linear, but we already return the string which by default is in english. but i think this must be done
@romaninsh @DarkSide666 what do you think?

Reply:
cannot be false if we make $this->fallback=en

need to add an exception in setLanguage or move the set of language and fallback directly in costructor to avoid throwing, because language must be defined only ones.

Reply:
Then just don't allow false as a value for fallback.
You've already listed the type as string, and it makes way more sense to simply have it as 'en' rather than false.
There's always going to be some text, in the worst case scenario you can simply return the original string.

No need for an exception, I think, but you could be missing some code that 'allows' languages.
Since now, there's no way to know what is valid and what is not, so you must assume that any string is a valid language.